### ISSUE TITLE
elimination

### ISSUE THREAD

**Issue by fieker** [oscar-system/Oscar.jl#2590]
add new interface using preimage to return ideal in small ring

**Comment by fingolfin** [comment]
@fieker it is not quite clear to me what you are asking here.

@HechtiDerLachs made a plausible guess: that `eliminate` should return its result in a new ring with fewer variables, not the original ring. Is that it?

**Comment by fingolfin** [comment]
Ah, the suggestion of using `preimage` probably means an interface involving a homomorphism and elimination is done via `preimage` computation.

**Comment by fingolfin** [comment]
@thofma suggests we may have this or something like this already in some sense? It'd be great if @thofma and @HechtiDerLachs could have a look and give us a short example how to use it here (and if it is, then perhaps this "just" needs someone to mention in the `eliminate` docstring...)

Independently from the above @micjoswig points out a relation to feature request #4650, suggesting some more links etc.

**Comment by thofma** [comment]
I don't think we have this. What we have is that we recognize morphisms of the form `f = hom(Qxy, Qxyz, [x, y])` (thanks to work of @HechtiDerLachs ).

At the moment we exploit this when evaluating `f(x)`. But we could also use it in the `preimage(f, I::Ideal)` code, where we would call `eliminate`. Then one clean interface for eliminating (and removing variables) would be to construct the `f` and then call `preimage(f, I)`.
